x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/tree/NodeSubclassTests.java

```diff
@@ -6,713 +6,21 @@
-import org.elasticsearch.xpack.esql.core.expression.function.Function;
-import org.elasticsearch.xpack.esql.core.expression.predicate.fulltext.FullTextPredicate;
-import org.elasticsearch.xpack.esql.core.expression.predicate.operator.comparison.In;
-import org.elasticsearch.xpack.esql.core.expression.predicate.regex.Like;
-import org.elasticsearch.xpack.esql.core.expression.predicate.regex.LikePattern;
-import org.elasticsearch.xpack.esql.core.tree.NodeTests.ChildrenAreAProperty;
-import org.elasticsearch.xpack.esql.core.tree.NodeTests.Dummy;
-import org.elasticsearch.xpack.esql.core.tree.NodeTests.NoChildren;
-import org.mockito.exceptions.base.MockitoException;
-
-import java.io.IOException;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
-import java.lang.reflect.ParameterizedType;
-import java.lang.reflect.Type;
-import java.lang.reflect.WildcardType;
-import java.nio.file.FileVisitResult;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.SimpleFileVisitor;
-import java.nio.file.attribute.BasicFileAttributes;
-import java.time.ZoneId;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.EnumSet;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Set;
-import java.util.function.Predicate;
-import java.util.function.Supplier;
-import java.util.jar.JarEntry;
-import java.util.jar.JarInputStream;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.emptyList;
-import static org.mockito.Mockito.mock;
+import java.util.function.Function;
 
 /**
- * Looks for all subclasses of {@link Node} and verifies that they
- * implement {@link Node#info()} and
- * {@link Node#replaceChildren(List)} sanely. It'd be better if
- * each subclass had its own test case that verified those methods
- * and any other interesting things that that they do but we're a
- * long way from that and this gets the job done for now.
- * <p>
- * This test attempts to use reflection to create believeable nodes
- * and manipulate them in believeable ways with as little knowledge
- * of the actual subclasses as possible. This is problematic because
- * it is possible, for example, for nodes to stackoverflow because
- * they <strong>can</strong> contain themselves. So this class
- * <strong>does</strong> have some {@link Node}-subclass-specific
- * knowledge. As little as I could get away with though.
- * <p>
- * When there are actual tests for a subclass of {@linkplain Node}
- * then this class will do two things:
- * <ul>
- * <li>Skip running any tests for that subclass entirely.
- * <li>Delegate to that test to build nodes of that type when a
- * node of that type is called for.
- * </ul>
+ * Shim to expose protected methods to ESQL proper's NodeSubclassTests.
```

Comment:
This was failing after deleting the unused subclasses. I figured it wasn't worth digging into why and instead merged as much of it as possible into `EsqlNodeSubclassTests`.